### DISCUSSION TITLE
Plugin API Request: expose Ride.type name string through the API

### DISCUSSION THREAD

**Discussion by ltsSmitty** [OpenRCT2/OpenRCT2#16579] [Feature requests]
I'd like to filter my rides by type, so that I can get a list of all Looping Coasters, Corkscrew Coasters, etc in my park. Currently I can use `map.rides[i].type`, and this returns a numeric type value (e.g. 59 if the ride is a Virginia Reel). 

I can't find a consistent map to show what ride type the returned numbers refer to. I attempted to use the [Ride Object C++ file > RideTypeLookupTable ](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/object/RideObject.cpp) and manually map it over myself, but the number didn't match up, giving me results like: 
`'ride name: Circus 1, ride type: 49, expect type from C++ lookup: Wooden Rollercoaster'`
`'ride name: Crooked House 1, ride type: 71, expect type from C++ lookup: Magic Carpet'`

I'd love a way to get the ride type name string corresponding to the ride type number. 

Thanks!

**Comment by ZehMatt** [comment]
You may have an error in your C++ lookup table, 71 = RIDE_TYPE_CROOKED_HOUSE (71) and not RIDE_TYPE_MAGIC_CARPET (77)

**Reply by ZehMatt** [reply]
Actual table: https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/Ride.h#L471

**Comment by autosysops** [comment]
I also got to this problem. I now fixed it by having a hard coded list in my plugin:
https://github.com/autosysops/OpenRct2RideEditor/blob/main/RideEditor.js

but this will not be translated. It would be nice if the translated string is available in the API.

**Comment by IntelOrca** [comment]
Ride types will at some point be objects, so it will be a case of getting the object using the ride type id and getting the name from that. You can already get the name for the vehicle object for a ride.
